entity-framework/core/querying/sql-queries.md

```diff
@@ -0,0 +1,163 @@
+
+The following example passes a single parameter to a stored procedure by including a parameter placeholder in the SQL query string and providing an additional argument:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureParameter)]
+
+While this syntax may look like regular C# [string interpolation](https://learn.microsoft.com/dotnet/csharp/language-reference/tokens/interpolated), the supplied value is wrapped in a `DbParameter` and the generated parameter name inserted where the `{0}` placeholder was specified. This makes > The <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> safe from SQL injection attacks, and sends the value efficiently and correctly to the database.
+
+When executing stored procedures, it can be useful to use named parameters in the SQL query string, especially when the stored procedure has optional parameters:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureNamedSqlParameter)]
+
+If you need more control over the database parameter being sent, you can also construct a `DbParameter` and supply it as a parameter value. This allows you to set the precise database type of the parameter, or facets such as its size, precision or length:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureSqlParameter)]
+
+> [!NOTE]
+>
+> Pay attention to the ordering of your parameters when executing a stored procedure. Unless using named parameters, the parameter in the SQL must exactly match the stored procedure definition, otherwise you may encounter type conversion exceptions and/or unexpected behavior when the procedure is executed.
+
+### Dynamic SQL and parameters
+
+<xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> and its parameterization should be used wherever possible. However, there are certain scenarios where SQL needs to be dynamically pieced together, and database parameters cannot be used. For example, let's assume that a C# variable holds the name of the a property to be filtered by. It may be tempting to use a SQL query such as the following:
+
+```c#
+var propertyName = "User";
+var propertyValue = "johndoe";
+
+var blogs = context.Blogs
+    .FromSql($"SELECT * FROM [Blogs] WHERE {propertyName} = {propertyValue}")
+    .ToList();
+```
+
+This code doesn't work, since database do not allow parameterizing column names (or any other part of the schema).
+
+First, carefully consider whether you really want to fully parameterize the column name. For example, users may choose a column that isn't indexed, making the query run extremely slowly and overload your database. Except for truly dynamic scenarios, it's usually better to have two SQL queries for two column names, rather than using parameterization to collapse them into a single SQL query.
```

Comment:
Not sure this really belongs here; the same is true when using the more common LINQ method builder to choose a property dynamically. I.e. it's not a consequence of using raw SQL. But I don't feel strongly.

Reply:
It's true that you have the same consideration with EF.Property - assuming you pass it arbitrary input, and not a fixed string (e.g. because of a shadow property - which I think is what it's usually used for).

The idea here was to make the user fully understand what it is they're doing when they interpolate a fully dynamic column name into their SQL (and possibly dissuade them from it, unless it's really required)... I do agree that a similar note could be a good idea for EF.Property as well.

I still think it's not a bad fit here to help "overall understanding/context", but can remove if people are against it.

Reply:
Revised this section to make it clear that it's a general note on dynamic queries rather than specific to SQL queries.